File: python/triangle/triangle.py
def is_equilateral(sides):
    return check_sides(sides, [1])


def is_isosceles(sides):
    return check_sides(sides, [1,2])


def is_scalene(sides):
    return check_sides(sides, [3])


def check_sides(sides, unique_sides):
    sides.sort()
    return sides[0] != 0 and \
        (sides[0] + sides[1] >= sides[2]) and \
        len(set(sides)) in unique_sides

File: ruby/triangle/triangle.rb
class Triangle
  def initialize(sides)
    @sides = sides.sort
  end

  def equilateral?
    check_sides([1])
  end

  def isosceles?
    check_sides([1,2])
  end

  def scalene?
    check_sides([3])
  end

  private
    attr_reader :sides

    def check_sides(unique_sides)
      sides[0] != 0 &&
      sides[0] + sides[1] >= sides[2] &&
      unique_sides.include?(sides.uniq.length)
    end
end
